website/docs/cloud-docs/integrations/kubernetes: Update API reference (#766)
hashicorp/terraform-docs-common: website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx

| --- | --- |
| `replicas` _integer_ |  |
| `spec` _[PodSpec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#podspec-v1-core)_ |  |

> | `annotations` _object (keys:string, values:string)_ | Annotations that will be applied to the pod template in the deployment. |
> | `labels` _object (keys:string, values:string)_ | Labels that will be applied to the pod template in the deployment. |

Reviewer: ```suggestion
| `annotations` _object (keys:string, values:string)_ | The annotations that the operator will apply to the pod template in the deployment. |
| `labels` _object (keys:string, values:string)_ | The labels that the operator will apply to the pod template in the deployment. |
```

| `stateVersions` _[WorkspaceStateVersionsPermissionType](#workspacestateversionspermissiontype)_ | State access. Must be one of the following values: `none`, `read`, `read-outputs`, `write`. Default: `none`. |
| `variables` _[WorkspaceVariablesPermissionType](#workspacevariablespermissiontype)_ | Variable access. Must be one of the following values: `none`, `read`, `write`. Default: `none`. |

#### DeletionPolicy

_Underlying type:_ _string_

> DeletionPolicy defines the strategies for resource deletion in the Kubernetes operator.
> It controls how the operator should handle the deletion of resources when triggered by
> a user action or system event.

Reviewer: ```suggestion
DeletionPolicy defines the strategy the Kubernetes operator uses when you delete a resource, either manually or by a system event.
```

> There are four possible values:

Reviewer: ```suggestion
You must use one of the following values:
```

> - `retain`: When the custom resource is deleted, the associated workspace is retained.
> - `soft`: Attempts to delete the associated workspace only if it does not contain any managed resources.
> - `destroy`: Executes a destroy operation to remove all resources managed by the associated workspace. Once the destruction of these resources is successful, the workspace itself is deleted, followed by the removal of the custom resource.
> - `force`: Forcefully and immediately deletes the workspace and the custom resource.

Reviewer: ```suggestion
- `retain`: When you delete the custom resource, the operator does not delete the workspace.
- `soft`: Attempts to delete the associated workspace only if it does not contain any managed resources.
- `destroy`: Executes a destroy operation to remove all resources managed by the associated workspace. Once the destruction of these resources is successful, the operator deletes the workspace, and then deletes the custom resource.
- `force`: Forcefully and immediately deletes the workspace and the custom resource.
```

| `sshKey` _[SSHKey](#sshkey)_ | SSH key used to clone Terraform modules. [More information](/terraform/cloud-docs/workspaces/settings/ssh-keys). |
| `notifications` _[Notification](#notification) array_ | Notifications allow you to send messages to other applications based on run and workspace events. [More information](/terraform/cloud-docs/workspaces/settings/notifications). |
| `project` _[WorkspaceProject](#workspaceproject)_ | Projects let you organize your workspaces into groups. Default: default organization project. [More information](/terraform/tutorials/cloud/projects). |

> | `deletionPolicy` _[DeletionPolicy](#deletionpolicy)_ | The Deletion Policy specifies the behavior of the custom resource and its associated workspace when the custom resource is deleted.<br />- `retain`: When the custom resource is deleted, the associated workspace is retained.<br />- `soft`: Attempts to delete the associated workspace only if it does not contain any managed resources.<br />- `destroy`: Executes a destroy operation to remove all resources managed by the associated workspace. Once the destruction of these resources is successful, the workspace itself is deleted, followed by the removal of the custom resource.<br />- `force`: Forcefully and immediately deletes the workspace and the custom resource.<br />Default: `retain`. |

Reviewer: ```suggestion
| `deletionPolicy` _[DeletionPolicy](#deletionpolicy)_ | The Deletion Policy specifies the behavior of the custom resource and its associated workspace when the custom resource is deleted.<br />- `retain`: When you delete the custom resource, the operator does not delete the workspace.<br />- `soft`: Attempts to delete the associated workspace only if it does not contain any managed resources.<br />- `destroy`: Executes a destroy operation to remove all resources managed by the associated workspace. Once the destruction of these resources is successful, the operator deletes the workspace, and then deletes the custom resource.<br />- `force`: Forcefully and immediately deletes the workspace and the custom resource.<br />Default: `retain`. |
```
